pkg/controller/atlasdeployment/serverless_deployment.go

```diff
@@ -40,8 +45,26 @@ func ensureServerlessInstanceState(ctx *workflow.Context, project *mdbv1.AtlasPr
 
 	switch atlasDeployment.StateName {
 	case status.StateIDLE:
+		convertedDeployment, err := serverlessSpec.ToAtlas()
+		if err != nil {
+			return atlasDeployment, workflow.Terminate(workflow.Internal, err.Error())
+		}
+		if convertedDeployment.Tags == nil {
+			convertedDeployment.Tags = &[]*mongodbatlas.Tag{}
+		}
+		if !isTagsEqual(*(atlasDeployment.Tags), *(convertedDeployment.Tags)) {
```

Comment:
Are we sure here that `atlasDeployment.Tags` never becomes nil?

Reply:
I believe the behaviour is that  `atlasDeployment.Tags` cannot go to nil after an initial value is fed to it, be it a full or empty array of pointers.
Since `performUpdate` passes after removing tags [here](https://github.com/mongodb/mongodb-atlas-kubernetes/blob/CLOUDP-188104/deployment-resource-tagging/test/int/deployment_test.go#L1101-L1103) that means AKO returns to idle so `atlasDeployment.Tags` has remained an empty array instead of converting to nil.